collect_app/src/androidTest/java/org/odk/collect/android/feature/smoke/BadServerTest.java

```diff
@@ -0,0 +1,65 @@
+package org.odk.collect.android.feature.smoke;
+
+import android.Manifest;
+import android.webkit.MimeTypeMap;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.GrantPermissionRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.injection.config.AppDependencyModule;
+import org.odk.collect.android.openrosa.OpenRosaHttpInterface;
+import org.odk.collect.android.support.CollectTestRule;
+import org.odk.collect.android.support.ResetStateRule;
+import org.odk.collect.android.support.StubOpenRosaServer;
+import org.odk.collect.android.support.pages.MainMenuPage;
+import org.odk.collect.utilities.UserAgentProvider;
+
+@RunWith(AndroidJUnit4.class)
+public class BadServerTest {
+
+    public final StubOpenRosaServer server = new StubOpenRosaServer();
+
+    public CollectTestRule rule = new CollectTestRule();
+
+    @Rule
+    public RuleChain copyFormChain = RuleChain
+            .outerRule(GrantPermissionRule.grant(
+                    Manifest.permission.READ_EXTERNAL_STORAGE,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                    Manifest.permission.READ_PHONE_STATE,
+                    Manifest.permission.GET_ACCOUNTS
+            ))
+            .around(new ResetStateRule(new AppDependencyModule() {
+                @Override
+                public OpenRosaHttpInterface provideHttpInterface(MimeTypeMap mimeTypeMap, UserAgentProvider userAgentProvider) {
+                    return server;
+                }
+            }))
+            .around(rule);
+
+    @Test
+    public void whenHashNotIncludedInFormList_canGetBlankForm_fillItIn_andSubmit() {
```

Comment:
I'd initially gone down the wrong path of trying to write a test for "legacy" servers (see `OpenRosaFormSource#parseLegacyFormList`) as the crash logs were showing that there was an Open Rosa version mismatch. It turns out that that still results in us parsing the form list as an Open Rosa response however so this case (an Open Rosa server not including hashes) is probably the main culprit here.